client/landing/gutenboarding/stores/domain-suggestions/selectors.ts

```diff
@@ -1,15 +1,63 @@
+/**
+ * External dependencies
+ */
+import { select } from '@wordpress/data';
+
 /**
  * Internal dependencies
  */
+import { STORE_KEY } from '.';
 import { DomainSuggestionQuery } from './types';
 import { State } from './reducer';
-import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+import { stringifyDomainQueryObject } from './utils';
 
 export const getState = ( state: State ) => state;
-export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
-	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
-	if ( serializedQuery ) {
-		return state.domainSuggestions[ serializedQuery ];
-	}
-	return null;
+
+type DomainSuggestionSelectorOptions = Partial< Exclude< DomainSuggestionQuery, 'query' > >;
+export const getDomainSuggestions = (
+	state: State,
+	search: string,
+	options: DomainSuggestionSelectorOptions = {}
+) => {
+	const normalizedQuery = normalizeDomainSuggestionQuery( search, options );
+
+	// We need to go through the `select` store to get the resolver action
+	return select( STORE_KEY ).__internalGetDomainSuggestions( normalizedQuery );
+};
+
+/**
+ * Do not use this selector. It is for internal use.
+ *
+ * @private
+ */
+export const __internalGetDomainSuggestions = (
+	state: State,
+	queryObject: DomainSuggestionQuery
+) => {
+	return state.domainSuggestions[ stringifyDomainQueryObject( queryObject ) ];
 };
+
+/**
+ * Normalize domain query
+ *
+ * It's important to have a consistent, reproduceable representation of a domains query so that the result can be
+ * stored and retrieved.
+ *
+ * @see client/state/domains/suggestions/utils.js
+ * @see client/components/data/query-domains-suggestions/index.jsx
+ */
+function normalizeDomainSuggestionQuery(
+	search: string,
+	q: DomainSuggestionSelectorOptions
+): DomainSuggestionQuery {
+	return Object.assign(
```

Comment:
Can we use object destructuring instead?

Also, I was kinda expecting a bit of a different syntax for the defaults handling, where the defaults are defined as an object into which our user-passed one is then merged, vaguely:

```js
const defaults = {
	include_wordpressdotcom: false,
	quantity: 5,
	vendor: 'variation2_front'
}

return {
	...defaults,
	...q,
	query: search.toLocaleLowerCase(),
}
```

which I personally find a bit easier on the eye. (If you want to strip unwanted properties from `q`, you could maybe add an extra `pick` call? Would also keep the concerns of sanitization and defaults separate.)

Reply:
(I think this would also mean we wouldn't need `??` for now and could decouple this PR entirely from that prereq.)

Reply:
I pushed something like this and rolled back the `??` usage 👍